src/bootstrap/builder.rs

```diff
@@ -926,7 +926,8 @@ impl<'a> Builder<'a> {
             cargo.env("RUSTC_ERROR_FORMAT", error_format);
         }
         if cmd != "build" && cmd != "check" && cmd != "rustc" && want_rustdoc {
-            cargo.env("RUSTDOC_LIBDIR", self.rustc_libdir(compiler));
+            // sysroot libdir required for case with custom libdir
```

Comment:
This seems suspicious. I would expect that we'd need to do a similar modification for `RUSTC_LIBDIR`, but that doesn't seem to be the case AFAICT.

A comment explaining why rustdoc should be different than rustc here would be good; it might be best to discuss that reasoning before adding the comment though since I'm not sure what it could be.

To clarify why this concerns me: I expect rustdoc and rustc to need/require the same libraries, so if one needs the sysroot/lib directory then the other should too, rather than rustc using sysroot/lib and rustdoc using sysroot/lib/rustlib/<triple>/lib. The libraries in both places are the same in stage 2 currently so this might "work" in that stage on accident, but I wouldn't expect it to in stage 1.

Reply:
It wasn't first time when I have to use this path for rustdoc library: https://github.com/rust-lang/rust/pull/52439
https://github.com/rust-lang/rust/pull/46592
Without it build fails even for host target.

Reply:
Right, I understand that this fixes it in your case (and maybe on CI). However, without understanding what goes wrong with your machine/config that causes that failure, and why this fixes it, I don't feel comfortable landing such a change. At the very least because it would seem reasonable to apply the same change to RUSTC_LIBDIR, too...

Reply:
@Mark-Simulacrum What if use here stage2 compile for stage2 rustdoc? Then correct librustc_driver library should match with RUSTDOC_LIBDIR.